src/SFA.DAS.Activities/Extensions/ActivityExtensions.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using SFA.DAS.Activities.Models;
+using SFA.DAS.NLog.Logger;
+
+
+namespace SFA.DAS.Activities.Extensions
+{
+    public static class ActivityExtensions
+    {
+        private static readonly ILog Log = new NLogLogger(typeof(ActivityExtensions));
+
+        public static ActivityUrlLink GetLinkDetails(this Activity activity)
+        {
+            var link = activity.Type.GetActivityLink();
+
+            if (link == null)
+                return null;
+
+            if (activity.Type == ActivityType.PaymentCreated)
```

Comment:
This does not follow the open/closed principle of SOLID.

Reply:
When I wrote the code I wasn't 100% sure if this was the best approach but followed the same approach as the activityType extension class. I agree that this is doing too much now for an extension so will change it.

Reply:
This has now been updated to a helper